metakube/resource_node_deployment_structure.go

```diff
@@ -299,23 +299,46 @@ func metakubeNodeDeploymentExpandSpec(p []interface{}) *models.NodeDeploymentSpe
 		return obj
 	}
 
+	var replicas *int32
+	if v, ok := in["replicas"]; ok && obj.MinReplicas == 0 {
+		if vv, ok := v.(int); ok {
+			replicas = int32ToPtr(int32(vv))
+		}
+	}
+
+	var min_replicas int32
 	if v, ok := in["min_replicas"]; ok {
 		if vv, ok := v.(int); ok {
-			obj.MinReplicas = int32(vv)
-			obj.Replicas = int32ToPtr(obj.MinReplicas)
+			min_replicas = int32(vv)
 		}
 	}
 
+	var max_replicas int32
 	if v, ok := in["max_replicas"]; ok {
 		if vv, ok := v.(int); ok {
-			obj.MaxReplicas = int32(vv)
+			max_replicas = int32(vv)
 		}
 	}
 
-	if v, ok := in["replicas"]; ok && obj.MinReplicas == 0 {
-		if vv, ok := v.(int); ok {
-			obj.Replicas = int32ToPtr(int32(vv))
+	if max_replicas > 0 {
+		obj.MinReplicas = min_replicas
+		obj.MaxReplicas = max_replicas
+
+		// Make sure replicas is between min and max
+		// If replicas is not set, we assume it is 0
+		// This way it will still get set between min and max
+		if replicas == nil {
+			replicas = int32ToPtr(0)
+		}
+		if *replicas > max_replicas {
+			obj.Replicas = int32ToPtr(max_replicas)
+		} else if *replicas < min_replicas {
+			obj.Replicas = int32ToPtr(min_replicas)
+		} else {
+			obj.Replicas = replicas
 		}
+	} else if replicas != nil {
+		obj.Replicas = replicas
 	}
 
```

Comment:
I may be missing something, but since if the Tf request only wants to change min/maxReplicas but not replicas, wouldn't this (still) set replicas to minReplicas?